pages/about/[slug].js

```diff
@@ -0,0 +1,39 @@
+import { MDXLayoutRenderer } from '@/components/MDXComponents'
+import { getFileBySlug } from '@/lib/mdx'
+import { getPostsByAuthor } from '@/lib/mdx'
+import { getAllAuthors } from '@/lib/mdx'
+
+export async function getStaticProps(context) {
+  let authorDetails = await getFileBySlug('authors', [context.params.slug])
+  const posts = await getPostsByAuthor(context.params.slug)
+  authorDetails.posts = posts
+  return { props: { authorDetails } }
+}
+
+export async function getStaticPaths() {
+  const authors = await getAllAuthors('blog')
+
+  var paths = []
+  for (let author in authors) {
+    paths.push({
+      params: { slug: author },
+    })
+  }
```

Comment:
mostly a nit about using var
```suggestion
  const paths = authors.map(author => ({ params: { slug: author } }))
```